### DISCUSSION TITLE
Converting Existing Certificates

### DISCUSSION THREAD

**Discussion by ponarangan** [airsdk/Adobe-Runtime-Support#1948] [Guides & Tutorials]
How to generate app signing certificate the existing apps on play store and how to create encryption key for app signing certificate for the already existing apps on play store? Pls help
@ajwfrost

**Comment by marchbold** [comment]
Your app signing certificate is the p12 file (or other) that you have been using with adt to package your apk / aab. When moving to Play Store signing you need to upload this certificate to the app signing section. 

You cannot generate a certificate for an existing app, you must use the same certificate you used to sign it in the past.

Is that what you are asking?

**Reply by ponarangan** [reply]
Adobe [Animate]
I want to update existing apps on play store and it is asking for app signing in. How to do play app signing with p.12 certificate using "Export and upload a key from java key store" option . I also want to know what is alias and encryption key and how to get them . Kindly help me.
![pep](https://user-images.githubusercontent.com/90509957/171339593-99cb7e14-dd14-418b-a742-dd03dca5cea1.jpg)

**Reply by marchbold** [reply]
Maybe have a look at this:

https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles#signing-and-certificates

**Reply by ponarangan** [reply]
Thanks. What is the difference between Enter password for store
Enter password for key

**Reply by marchbold** [reply]
Can you elaborate on that? If you are talking about a key inside a certificate then you just have to enter the same password.

**Reply by ponarangan** [reply]
When i Used the same password for store and key its shows error like this
![error](https://user-images.githubusercontent.com/90509957/171678633-105ee84f-455c-433e-b403-d3299b60f2dd.jpg)
.

**Reply by marchbold** [reply]
Is it an older 1024bit certificate and what version of java are you using? 

Newer versions of java (< 1.8.112) don't support the 1024 bit certificates so you may be having that issue?
